src/core/log_callback.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <string>
+#include <functional>
+
+namespace mavsdk::log {
+
+// defined numeric values can be useful for filtering message by severity
+typedef enum {
```

Comment:
Since we are changing this, what about an `enum class`? Feels more modern :blush:

Reply:
My bad, I didn't know enum class can have underlying type. Definitely a good idea, especially since this enum becomes public. I'll change it